### PR TITLE
Config: Implemented inheritance policy for autoReload in included config files

### LINKED ISSUE #668: When <include>ing a file with a <nlog> root element, autoReload in parent config is ignored
Discovered for version 3.1. Apologize for not having time to test newer versions, we're getting a release ready :P
Small (untested) example to illustrate:
NLog.config:
```xml
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  autoReload="true" internalLogToConsoleError="true">
  <include file="include.config" />
</nlog>
```
include.config:
```xml
    <nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    </nlog>
```

AutoReload will not be enabled.

Adding autoReload="true" to include.config enables autoReload for both files (which corresponds to documented behaviour).

### PR DESCRIPTION
**Pull request by bhaeussermann**
Fixes #668

### CODE REVIEW

File: src/NLog/Config/XmlLoggingConfiguration.cs [hunk 1]
```diff
@@ -180,14 +180,7 @@ public static LoggingConfiguration AppConfig
         /// Did the <see cref="Initialize"/> Succeeded? <c>true</c>= success, <c>false</c>= error, <c>null</c> = initialize not started yet.
         /// </summary>
         public bool? InitializeSucceeded { get; private set; }
-
-
-        /// <summary>
-        /// Gets or sets a value indicating whether the configuration files
-        /// should be watched for changes and reloaded automatically when changed.
-        /// </summary>
-        public bool AutoReload { get; set; }
```
Comment (304NotModified) [inline src/NLog/Config/XmlLoggingConfiguration.cs]: Can you tell me why this property is not needed anymore? it's now a breaking change!

File: src/NLog/Config/XmlLoggingConfiguration.cs [hunk 2]
```diff
@@ -383,22 +364,12 @@ private void CheckUnusedTargets()
             InternalLogger.Debug("Unused target checking is completed. Total Rule Count: {0}, Total Target Count: {1}, Unused Target Count: {2}", this.LoggingRules.Count, configuredNamedTargets.Count, unusedCount);
         }
 
-        private void ConfigureFromFile(string fileName)
+        private void ConfigureFromFile(string fileName, bool autoReloadDefault = false)
```
Comment (304NotModified) [inline src/NLog/Config/XmlLoggingConfiguration.cs]: I prefer to make parameters of private methods always required (no optional). 

You could not know that, there is not coding rules (yet)

**Comment by 304NotModified** [comment]
Thanks! Which unit tests are new now?

This is now a breaking change. We should try to get it non-breaking.

**Comment by bhaeussermann** [comment]
> Thanks! Which unit tests are new now?

`Auto_Reload_after_rename()` in `LogFactoryTests` has been replaced by `TestAutoReloadOnFileCopy()`.  
`Auto_reload_validxml_test()` and `Auto_reload_invalidxml_test()` have been replaced by `TestAutoReloadOnFileChange()`.

All the other tests are new.

**Comment by bhaeussermann** [comment]
@304NotModified I have applied your feedback:
* Added back the AutoReload property.
* Made all method parameters mandatory.

**Comment by 304NotModified** [comment]
Thanks! 

IMO arrays are legacy stuff (and evil). Can you replace `toArray` with `toList`? Then I will merge it :)

**Comment by bhaeussermann** [comment]
> IMO arrays are legacy stuff (and evil). Can you replace toArray with toList? Then I will merge it :) 

Done ;).

**Comment by 304NotModified** [comment]
NICE!

**Pull request merged**